src/irmin-pack/unix/append_only_file.ml

```diff
@@ -21,23 +21,19 @@ module Make (Io : Io.S) (Errs : Io_errors.S with module Io = Io) = struct
   module Io = Io
   module Errs = Errs
 
+  let auto_flush_threshold = 16_384
```

Comment:
I'm running a benchmark to see what impact this has.

Reply:
![stats](https://github.com/mirage/irmin/assets/87719817/b0de1b8a-9208-4b5e-9e5a-a4ee5b793b65)

I'm having issues generating the textual stats for the benchmark but the graph is clearly showing a slowdown. I suppose we should try running with the previous `1_000_000` to verify it is the cause.

Reply:
Ran a benchmark that was slower but then I remembered that 3.8 changed LRU semantics. I need to re-run with an adjusted LRU size.

Reply:
I ran the benchmark with adjusted LRU entry size. https://github.com/metanivek/irmin-tezos-benchmarking/tree/rm_auto_flush

I'm having issues with generating `stats.txt` for the benchmark (separate issue) but here are the results.

The first comparison is against 3.7.1 baseline that we've been using.
![stats](https://github.com/mirage/irmin/assets/87719817/3192609b-9c00-469e-b79c-d67eed8d0bdc)

This makes this PR look like it is a huge performance improvement. Well, then I remembered that the new LRU performs better when compared to 3.7.

So here is this PR compared to the equivalent run for the new LRU (15k entries).
![stats new_lru](https://github.com/mirage/irmin/assets/87719817/2d1e0ff5-d65f-44cc-bde9-79a0e1c249b9)

I don't know the exact numbers since I can't get the summary data yet, but I can't imagine it is much different, given how closely these lines are. I think this PR is good to go from a performance perspective!

Reply:
Thanks a ton for running the benchmarks! It's great to have confirmation that this PR doesn't have any impact as expected :)